src/services/quandl-service.js

```diff
@@ -55,17 +41,37 @@
                 });
             }
 
+            // Queries Quandl for all stocks matching the input query
             function search(query, cb) {
-                stockSearch().get({ query: query }, function(result) {
+                var deferred = $q.defer();
+                var httpConfig = {
+                    get: {
+                        method: 'GET',
+                        cache: true,
+                        transformResponse: function(data, headers) {
+                            var json = angular.fromJson(data);
+                            return filterByDate(json);
+                        }
+                    }
+                };
+
+                httpConfig.get.timeout = deferred.promise;
```

Comment:
Can we write at least a short comment here? Is this related to friendly message?